src/lib/core/CHIPError.h

```diff
@@ -159,10 +183,17 @@ class ChipError
      * @note
      *  This is intended to be used only in foreign function interfaces.
      */
+#if __cplusplus >= 202002L
+    explicit constexpr ChipError(StorageType error) : mError(error) CHIP_INITIALIZE_ERROR_SOURCE(nullptr, 0, std::source_location()) {}
```

Comment:
Please add a unit test for this behavior in the chip error case, and include ifdef guards for c++20